platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/internal/AbstractTaskSchedulerBuilder.java

```diff
@@ -146,6 +164,9 @@ public AbstractTaskSchedulerBuilder<OUT> withType(@NonNull final TaskSchedulerTy
     @Override
     @NonNull
     public AbstractTaskSchedulerBuilder<OUT> withUnhandledTaskCapacity(final long unhandledTaskCapacity) {
+        if (type == DIRECT || type == DIRECT_THREADSAFE) {
```

Comment:
I think it makes sense to call the method here as well, just to avoid duplication